### PR TITLE
Customize Registration Forms

### LINKED ISSUE #37: Android example app EULA agreement cut off
When running the example app on Android (Pixel 4 emulator API 29, Android Studio 4.1.1), the "I have read and agree to the Terms & Conditions" text gets cut off on the right side. I think this also existed prior to 2.x.

Screen Shot 2020-12-11 at 12 26 29 PM

### LINKED ISSUE #39: Submit form on enter
#### Describe the desired behavior

Forms shall all be submitted on tapping "done", such as the login screen, the password changing screen, etc.

#### Describe the current behavior

It only dismisses the keyboard.

### LINKED ISSUE #40: tapping "login" after both email addr and pwd has been entered correctly should log in directly
#### Describe the desired behavior

On the login screen, tapping on the "login" button after both the email address and the password has been entered correctly should log in directly.

#### Describe the current behavior

If you:
(0) toggle system keyboard
(1) enter valid email
(2) enter password
(3) without touching anywhere else, tap "login"

Observe that only the keyboard got dismissed. The user is not logged in.

### LINKED ISSUE #44: AsyncStorage has moved
#### Describe the bug
When installing dependencies there is a warning that says that AsyncStorage has moved to a new location.

#### What is the expected behavior?
No warnings, use the newer package location.

#### What are the steps to reproduce?
1. Clone the repo
2. Start the demo
3. Observe the terminal output

#### Anything else to add?
This should be as simple as swapping out the package in the dependencies and updating the import statement.

### PR DESCRIPTION
**Pull request by joebochill**
For PXBLUE-1878
For PXBLUE-1938
For PXBLUE-1936
Fixes #40 
Fixes #39 
Fixes #44 
Fixes #37 

#### Changes proposed in this Pull Request:
- Add the ability to inject custom forms into registration
- Add ability to customize the instructions and the title on injected form pages
- Add ability to inject custom content below the login button and above the registration links (`loginActions`)
- Clean up form submission on Enter/Submit/Done
- Update documentation

### CODE REVIEW

**Review by emclaug2 — CHANGES_REQUESTED** [review]
The custom account details forms look good to me, one thing I noticed is that in the `Create Account` workflow, the entered email is disregarded at the end.

image

**Comment by joebochill** [comment]
> The custom account details forms look good to me, one thing I noticed is that in the `Create Account` workflow, the entered email is disregarded at the end.

This information actually comes from the user's implementation of the Registration actions. In the final call to complete registration, they need to return back an object with the details of the account (this is because it's generic enough to handle invite registration or self registration, so WE might not know the email address that was registered):

https://github.com/pxblue/react-native-workflows/blob/feature/custom-registration/login-workflow/example/src/actions/RegistrationUIActions.tsx#L93

I can change it in the demo so it makes more sense, but ultimately it's up the the end user to provide that information.

**Comment by emclaug2** [comment]
I'm going to approve this PR since I didn't notice anything related to the account details customization.

**Review by emclaug2 — APPROVED** [review]

**Comment by jeffvg** [comment]
does loginAction have props to display 'Log In With Google'

**Comment by joebochill** [comment]
> does loginAction have props to display 'Log In With Google'
> 
> ok just JSX.element?

It's just JSX.Element so the user can pass in any content they want. If they want a google button, they pass in a google button, if they want a facebook button, they pass in a facebook button, etc.

**Comment by jeffvg** [comment]
> > does loginAction have props to display 'Log In With Google'
> > ok just JSX.element?
> 
> It's just JSX.Element so the user can pass in any content they want. If they want a google button, they pass in a google button, if they want a facebook button, they pass in a facebook button, etc.

ok maybe we can go over this later. I have this right now...
Property 'loginActions' does not exist on type 'AuthUIContextProviderProps'

File: login-workflow/example/App.tsx [hunk 1]
```diff
@@ -36,6 +37,10 @@ export const AuthUIConfiguration: React.FC = (props) => {
             contactEmail={'something@email.com'}
             contactPhone={'1-800-123-4567'}
             contactPhoneLink={'1-800-123-4567'}
+            customAccountDetails={[
+                { component: CustomAccountDetails },
+                { title: 'CustomTitle', instructions: 'Get your details in yo', component: CustomAccountDetailsTwo },
```
Comment (jeffvg) [inline login-workflow/example/App.tsx:42]: Looks like instructions: needs updated

**Pull request merged**